app/src/test/kotlin/com/wire/android/ui/WireActivityViewModelTest.kt

```diff
@@ -0,0 +1,231 @@
+import com.wire.android.config.TestDispatcherProvider
+import com.wire.android.config.mockUri
+import com.wire.android.navigation.NavigationCommand
+import com.wire.android.navigation.NavigationItem
+import com.wire.android.navigation.NavigationManager
+import com.wire.android.notification.WireNotificationManager
+import com.wire.android.util.deeplink.DeepLinkProcessor
+import com.wire.android.util.deeplink.DeepLinkResult
+import com.wire.kalium.logic.configuration.GetServerConfigResult
+import com.wire.kalium.logic.configuration.GetServerConfigUseCase
+import com.wire.kalium.logic.configuration.ServerConfig
+import com.wire.kalium.logic.data.id.ConversationId
+import com.wire.kalium.logic.data.user.UserId
+import com.wire.kalium.logic.feature.auth.AuthSession
+import com.wire.kalium.logic.feature.session.CurrentSessionFlowUseCase
+import com.wire.kalium.logic.feature.session.CurrentSessionResult
+import io.mockk.MockKAnnotations
+import io.mockk.coEvery
+import io.mockk.coVerify
+import io.mockk.every
+import io.mockk.impl.annotations.MockK
+import io.mockk.mockk
+import kotlinx.coroutines.ExperimentalCoroutinesApi
+import kotlinx.coroutines.flow.flowOf
+import org.amshove.kluent.internal.assertEquals
+import org.junit.jupiter.api.Test
+import org.junit.jupiter.api.extension.ExtendWith
+
+@OptIn(ExperimentalCoroutinesApi::class)
+@ExtendWith(CoroutineTestExtension::class)
+class WireActivityViewModelTest {
+
+    @Test
+    fun `given Intent is null, when currentSession is present, then startNavigation is Home`() {
+        val (_, viewModel) = Arrangement()
+            .withSomeCurrentSession()
+            .arrange()
+
+        viewModel.handleDeepLink(null)
+
+        val startDestination = viewModel.startNavigationRoute()
+        assertEquals(NavigationItem.Home.getRouteWithArgs(), startDestination)
+    }
+
+    @Test
+    fun `given Intent is null, when currentSession is absent, then startNavigation is Welcome`() {
+        val (_, viewModel) = Arrangement()
+            .withNoCurrentSession()
+            .arrange()
+
+        viewModel.handleDeepLink(null)
+
+        val startDestination = viewModel.startNavigationRoute()
+        assertEquals(NavigationItem.Welcome.getRouteWithArgs(), startDestination)
+    }
+
+    @Test
+    fun `given Intent with SSOLogin, when currentSession is present, then startNavigation is Login and navArguments contains SSOLogin`() {
+        val (_, viewModel) = Arrangement()
+            .withSomeCurrentSession()
+            .withDeepLinkResult(DeepLinkResult.SSOLogin.Success("cookie", "config"))
+            .arrange()
+
+        viewModel.handleDeepLink(mockedIntent())
+
+        assertEquals(NavigationItem.Login.getRouteWithArgs(), viewModel.startNavigationRoute())
+        assert(viewModel.navigationArguments().filterIsInstance<DeepLinkResult.SSOLogin>().isNotEmpty())
+    }
+
+    @Test
+    fun `given Intent with CustomServerConfig, then startNavigation is Login and no SSOLogin in navArguments`() {
```

Comment:
can you add the when block ? `when currentSession is present,`
Also can you cover this case with test ?
`given Intent with CustomServerConfig, when currentSession is absent, then startNavigation is Login and no SSOLogin in navArguments`